lib/rex/proto/nuuo/client.rb

```diff
@@ -0,0 +1,270 @@
+# -*- coding: binary -*-
+
+require 'rex/proto/nuuo/client_request'
+require 'rex/proto/nuuo/response'
+
+module Rex
+module Proto
+module Nuuo
+# This class is a representation of a nuuo client
+class Client
+  # @!attribute host
+  #   @return [String] The nuuo server host
+  attr_accessor :host
+  # @!attribute port
+  #   @return [Integer] The nuuo server port
+  attr_accessor :port
+  # @!attribute timeout
+  #   @return [Integer] The connect/read timeout
+  attr_accessor :timeout
+  # @!attribute protocol
+  #   @return [String] The transport protocol used (tcp/udp)
+  attr_accessor :protocol
+  # @!attribute connection
+  #   @return [IO] The connection established through Rex sockets
+  attr_accessor :connection
+  # @!attribute context
+  #   @return [Hash] The Msf context where the connection belongs to
+  attr_accessor :context
+  # @!attribute ncs_version
+  #   @return [String] NCS version used in session
+  attr_accessor :ncs_version
+  # @!attribute username
+  #   @return [String] Username for NCS
+  attr_accessor :username
+  # @!attribute password
+  #   @return [String] Password for NCS user
+  attr_accessor :password
+  # @!attribute user_session
+  #   @return [String] ID for the user session
+  attr_accessor :user_session
+  # @!attribute config
+  #   @return [Hash] ClientRequest configuration options
+  attr_accessor :config
+
+  def initialize(opts = {})
+    self.host         = opts[:host]
+    self.port         = opts[:port] || 5180
+    self.timeout      = opts[:timeout] || 10
+    self.protocol     = opts[:protocol] || 'tcp'
```

Comment:
If we don't actually support anything other than TCP, do we need to really define it at all? I wonder if there is anything that implicitly assumes this (e.g. logging functions / peer)... Probably not!